modules/programs/neovim.nix

```diff
@@ -20,10 +20,16 @@ let
     options = {
       config = mkOption {
         type = types.lines;
-        description = "vimscript for this plugin to be placed in init.vim";
+        description = "Script to configure this plugin. The scripting language should match type.";
         default = "";
       };
 
+      type = mkOption {
+        type = types.str;
+        description = "Language used in config. For instance 'viml', 'lua', 'fennel', 'teal', ...";
+        default = "viml";
+      };
+
```

Comment:
We could retain compatibility with a list of derivations if this option is renamed to something like `language`.

Reply:
Also, using `types.either (types.enum [ "lua" "viml" "..." ]) types.str` provides for a good documentation experience :wink:

Reply:
at first I had called it "configType" then renamed it to `type` and then wondered if "language" or "lang" wouldn't be better (as it is recurrent, better to keep the name simple ?).
`types.either (types.enum [ "lua" "viml" "..." ]) types.str` was also something I pondered with. Glad you suggest it, I will add it

Reply:
the name of the option is my last blocking point for merging.

Reply:
I don't really mind the name, though `type` has the advantage of being shorter to type :smile:

Reply:
that was my reasoning too ^^